design/ceph-config-updates.md

```diff
@@ -0,0 +1,127 @@
+need for having a `ceph.conf` in containers at all. In practice, we do not know how to make this an
+actual possibility.
+
+The troublesome situation regards Ceph daemons' ability to discover mons. The daemons must have
+`mon host` specified to find mons which are using non-default ports. It is further undesirable to
+specify this via the `--mon-host` flag because if this parameter is specified statically as a
+container argument and the initial mons fail and are replaced by new mons, the daemon will not know
+how to contact the mons if it is restarted.
+
+The proposed solution to this problem and the only config option which have thus far been deemed
+*required* is the `mon host` global option which will be stored in a Kubernetes ConfigMap so that
+containers always have the most updated version of this information. The single `ceph.conf`
+ConfigMap will be mounted to all containers at `/etc/ceph/ceph.conf`. There is a possibility the
+`mon initial members` global option may also be necessary, as it goes hand-in-hand with `mon host`.
+
+With this one -- or these two -- options set via the config file, all daemons should *require* no
+other config file options set.
+
+#### Convenient minimum
+
+There may be a small set of additional options which are okay to have in the config file for
+convenience if they would otherwise be set via command line for every single daemons.
+
+For example:
+- `log file = /dev/stderr` should be set for all daemons to enforce the container standard of
+  sending logging to stdout/stderr.
+- `mon cluster log file = /dev/stderr` also for reasons above
+- `log stderr prefix = "debug "` should be set for all daemons to differentiate logging from auditing
+  - TODO: should this be set via Ceph centralized config if someone wants to change this? What about Luminous?
+- TODO: did I miss any?
+- `fatal signal handlers = false`
+  **TODO MORE INFORMATION NEEDED** why does Rook set this? It's purpose will determine its config location
+
+#### Additional configs for test environments
+
+Test (especially integration tests) may need to specify `osd pool default size = 1` and
+`osd pool default min size = 1` to support running clusters with only one osd. Test environments
+would have a means of doing this fairly easily using the still-existent config override capability.
+
+### Minimal command line options
+
+The minimal set of command line options is pared down to the settings which inform the daemon where
+to find and/or store data.
+
+Required flags:
+- `--keyring` may necessary to inform daemons where to find their keyrings
+- `--{mon,osd,mgr,mds,rgw}-data-dir` settings exist for all daemons, and these will continue to need
+  set since the locations Rook uses are non-default
+- `--run-dir` will need set on the command line since it is non-default and no longer set in the
+  config file
+
+Notable non-required flags:
+- `--fsid` does not need to be set, as daemons can learn this from their mons
```

Comment:
it's needed for the bootstrap phase of the mons (i.e., by the init container that runs ceph-mon --mkfs ...).

it can be passed as a command line arg to every other daemon as a safety check so that teh daemon will not accidentally connect to a different ceph cluster.